nucypher/blockchain/eth/events.py

```diff
@@ -100,13 +97,15 @@ def __init__(self,
             raise ValueError(f"Contract agent must be provided")
         if not event_name:
             raise ValueError(f"Event name must be provided")
+        if not from_block:
+            raise ValueError(f"From block must be provided")
 
         self.event_filter = agent.events[event_name]
         self.from_block = from_block
         self.to_block = to_block if to_block else agent.blockchain.client.block_number
         # validity check of block range
-        if to_block <= from_block:
-            raise ValueError(f"Invalid block range provided ({from_block} - {to_block})")
+        if self.to_block <= self.from_block:
+            raise ValueError(f"Invalid block range provided ({self.from_block} - {self.to_block})")
```

Comment:
May be better reported as "Final block must be greater than the starting block" or something.

Reply:
Also, while it's not clear from `web3` documentations, [this SE link](https://ethereum.stackexchange.com/questions/78620/is-the-range-of-a-filterquery-inclusive-up-to-toblock) seems to indicate that the range is inclusive. So `to_block == from_block` is still a valid range.